test/irb/helper.rb

```diff
@@ -73,4 +78,109 @@ def without_rdoc(&block)
       }
     end
   end
+
+  class IntegrationTestCase
+    LIB = File.expand_path("../../lib", __dir__)
+    TIMEOUT_SEC = 3
+
+    def setup
+      unless defined?(PTY)
+        omit "Integration tests require PTY."
+      end
+    end
+
+    def run_ruby_file(&block)
+      cmd = [EnvUtil.rubybin, "-I", LIB, @ruby_file.to_path]
+      tmp_dir = Dir.mktmpdir
+
+      @commands = []
+      lines = []
+
+      yield
+
+      PTY.spawn(integration_envs.merge("TERM" => "dumb"), *cmd) do |read, write, pid|
+        Timeout.timeout(TIMEOUT_SEC) do
+          while line = safe_gets(read)
+            lines << line
+
+            # means the breakpoint is triggered
+            if line.match?(/binding\.irb/)
+              while command = @commands.shift
+                write.puts(command)
+              end
+            end
+          end
+        end
+      ensure
+        read.close
+        write.close
+        kill_safely(pid)
+      end
+
+      lines.join
+    rescue Timeout::Error
+      message = <<~MSG
+      Test timedout.
+
+      #{'=' * 30} OUTPUT #{'=' * 30}
+        #{lines.map { |l| "  #{l}" }.join}
+      #{'=' * 27} END OF OUTPUT #{'=' * 27}
+      MSG
+      assert_block(message) { false }
+    ensure
+      File.unlink(@ruby_file) if @ruby_file
+      FileUtils.remove_entry tmp_dir
+    end
+
+    # read.gets could raise exceptions on some platforms
+    # https://github.com/ruby/ruby/blob/master/ext/pty/pty.c#L729-L736
```

Comment:
Line number is changed now. https://github.com/ruby/ruby/blob/master/ext/pty/pty.c#L721-L728

Reply:
Updated 👍